paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt

```diff
@@ -51,21 +56,43 @@ import kotlin.io.path.invariantSeparatorsPathString
 class PaparazziPlugin : Plugin<Project> {
   override fun apply(project: Project) {
     project.afterEvaluate {
-      check(!project.plugins.hasPlugin("com.android.application")) {
-        error(
-          "Currently, Paparazzi only works in Android library -- not application -- modules. " +
-            "See https://github.com/cashapp/paparazzi/issues/107"
-        )
-      }
-      check(project.plugins.hasPlugin("com.android.library")) {
-        "The Android Gradle library plugin must be applied for Paparazzi to work properly."
+      val newResourceLoadingEnabled =
+        (project.findProperty("app.cash.paparazzi.new.resource.loading") as? String)?.toBoolean() ?: false
+
+      if (!newResourceLoadingEnabled) {
+        check(!project.plugins.hasPlugin("com.android.application")) {
+          error(
+            "Currently, Paparazzi only works in Android library -- not application -- modules. " +
+              "See https://github.com/cashapp/paparazzi/issues/107"
+          )
+        }
+
+        check(project.plugins.hasPlugin("com.android.library")) {
+          "The Android Gradle library plugin must be applied for Paparazzi to work properly."
+        }
+
+        project.plugins.withId("com.android.library") {
+          setupPaparazzi(project, project.extensions.getByType(LibraryExtension::class.java).libraryVariants)
+        }
+      } else {
+        check(listOf("com.android.application", "com.android.library", "com.android.dynamic-feature").any { project.plugins.hasPlugin(it) }) {
+          "The Android Gradle application or library or dynamic feature plugin must be applied for Paparazzi to work properly."
+        }
```

Comment:
any suggestion for better description?

Reply:
I would extract list from `check` and say:
"One of ${list.joinToString(", ")} must be applied for Paparazzi to work properly.

That way users don't have to map from "name in message" to "actual.plugin.id" in their head.